core/trino-main/src/main/java/io/trino/type/LikeFunctions.java

```diff
@@ -76,6 +76,22 @@ public static LikeMatcher likePattern(@SqlType("varchar") Slice pattern, @SqlTyp
         }
     }
 
+    public static boolean isMatchAllPattern(Slice pattern, Optional<Slice> escape)
+    {
+        if (escape.isPresent()) {
+            return false;
+        }
```

Comment:
This should work with escapes as long as the % character is not escaped (i.e., treated as a literal)